phases/19-capstone-projects/47-checkpoint-save-resume/code/test_main.py

```
@@ -0,0 +1,122 @@
+class CheckpointResumeTests(unittest.TestCase):
+    def test_single_file_round_trip_matches_state(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            ckpt.seed_everything(0)
+            model, opt, sched = _build_components(total_steps=4)
+            state = ckpt.TrainState(step=3, epoch=1, batch_in_epoch=2, losses=[0.5, 0.4, 0.3])
+            target = Path(tmp) / "ckpt.pt"
+            ckpt.save_checkpoint(model, opt, sched, state, target)
+
+            ckpt.seed_everything(99)
+            model2, opt2, sched2 = _build_components(total_steps=4)
+            for (k, v1), (_, v2) in zip(model.state_dict().items(), model2.state_dict().items()):
```

Ruff (0.15.14)

[warning] Loop control variable `k` not used within loop body

(B007)